Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/measure.h

```diff
@@ -211,6 +211,63 @@ squared_edge_length(typename boost::graph_traits<PolygonMesh>::edge_descriptor e
   return squared_edge_length(halfedge(e, pmesh), pmesh, np);
 }
 
+/**
+  * \ingroup PMP_measure_grp
+  *
+  * returns the longest edge of a given polygon mesh.
+  *
+  * @tparam PolygonMesh a model of `HalfedgeListGraph`
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param pmesh the polygon mesh in which the longest edge is searched for 
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamNBegin{vertex_point_map}
+  *     \cgalParamDescription{a property map associating points to the vertices of `pmesh`}
+  *     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+  *                    as key type and `%Point_3` as value type}
+  *     \cgalParamDefault{`boost::get(CGAL::vertex_point, pmesh)`}
+  *   \cgalParamNEnd
+  *
+  *   \cgalParamNBegin{geom_traits}
+  *     \cgalParamDescription{an instance of a geometric traits class}
+  *     \cgalParamType{a class model of `Kernel`}
+  *     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+  *     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+  *   \cgalParamNEnd
+  * \cgalNamedParamsEnd
+  *
+  * @return the longest edge in `pmesh`. The return type is an `edge_descriptor`.
+  *
+  * @sa `squared_edge_length()`
+  * @sa `longest_border()`
+  */
+
+template<typename PolygonMesh,
+         typename NamedParameters = parameters::Default_named_parameters>
+inline typename boost::graph_traits<PolygonMesh>::edge_descriptor
+longest_edge(const PolygonMesh& pmesh, const NamedParameters& np = parameters::default_values()) {
+
+    auto edge_range = edges(pmesh);
+
+    // if mesh has no edges
+    if (edge_range.begin() == edge_range.end())
+        return boost::graph_traits<PolygonMesh>::null_edge();
+
+    auto edge_reference = std::max_element(edge_range.begin(), edge_range.end(), [&pmesh, &np](auto l, auto r) {
+        return squared_edge_length(l, pmesh, np)
```

Comment:
This should use [CompareDistance_3](https://doc.cgal.org/latest/Kernel_23/classKernel_1_1CompareDistance__3.html#a2ee64f4b6e1ac24eab6935efcf9d22ca)